app/src/main/java/com/example/playlistmaker/SettingsActivity.kt

```diff
@@ -15,8 +16,39 @@ class SettingsActivity : AppCompatActivity() {
         val backSettingsBtn = findViewById<androidx.appcompat.widget.Toolbar>(R.id.settings_back)
 
         backSettingsBtn.setOnClickListener {
-            val displayIntent = Intent(this, MainActivity::class.java)
-            startActivity(displayIntent)
+            finish()
+        }
+
+
+        val shareBtn = findViewById<com.google.android.material.textview.MaterialTextView>(R.id.share)
+
+        shareBtn.setOnClickListener {
+            val shareIntent = Intent(Intent.ACTION_SEND)
+            shareIntent.setType("text/plain");
+            shareIntent.putExtra(Intent.EXTRA_TEXT, R.string.share_url)
```

Comment:
```suggestion
            shareIntent.putExtra(Intent.EXTRA_TEXT, getString(R.string.share_url))
```
Нужно передавать не айди, а саму строку